### ISSUE TITLE
Memory leak during animation of GIF's in android

### ISSUE THREAD

**Issue by vaibhav-46** [facebook/fresco#976]
I consistently see memory leaks while integrating Fresco for playing Gifs in my app. The xml of the images is as follows : 

<Framelayout
   <RelativeLayout
     <TextView>
     <com.facebook.drawee.view.SimpleDraweeView>
    </RelativeLayout>
   <TextView>
</FrameLayout>

With the above XML layout, I then try and play a Gif using an asset file, using the code as follows :

``````
try {
            f = createFileFromInputStream(AndroidContext.getContext().getAssets().open(filepath));
        } catch (IOException e) {
            e.printStackTrace();
        }
        try {
            Uri uri = Uri.fromFile(f);
            DraweeController controller = Fresco.newDraweeControllerBuilder()
                    .setUri(uri)
                    .setAutoPlayAnimations(true)
                    .build();
            gificons.setController(controller);
            gificons.setHierarchy(hierarchy);
            gificons.setVisibility(View.VISIBLE);
            //gificons.setMinimumWidth(40);

            Animatable animatable = gificons.getController().getAnimatable();
            if (animatable != null) {
                animatable.start();
            }
            gificons.requestLayout();
            gificons.refreshDrawableState();
``

The Gif plays fine but there is a lot of memory leak almost at the rate of 1 MB/min. I added more logging to check what was happening and was getting the following error : ```


 82: initialize
02-04 22:13:41.226 17386-17386/? V/unknown:AbstractDraweeController: controller 42456e80 82: onDetach
02-04 22:13:41.226 17386-17386/? V/unknown:AbstractDraweeController: controller 42456e80 82: setHierarchy: null
02-04 22:13:41.226 17386-17386/? V/unknown:AbstractDraweeController: controller 42456e80 82: setHierarchy: com.facebook.drawee.generic.GenericDraweeHierarchy@42453d08
02-04 22:13:41.233 17386-17386/? V/unknown:AbstractDraweeController: controller 42456e80 82: onAttach: request needs submit
02-04 22:13:41.233 17386-17386/? V/unknown:AbstractDraweeController: controller 42456e80 82: submitRequest: dataSource: 43b4e860
02-04 22:13:41.233 17386-17386/? V/unknown:AbstractDraweeController: controller 42456e80 82: final_failed @ onFailure: failure: java.lang.NullPointerException: No image request was specified!

The Gif animates fine but there is a lot of memory leak and it slowly builds up.
``````

``

**Comment by massimocarli** [comment]
Thank you for the detailed description. We're already working on that.

Thanks

**Comment by e-hai** [comment]
do it fix?

**Comment by oprisnik** [comment]
We've released [Fresco v1.4.0](https://github.com/facebook/fresco/releases/tag/v1.4.0), which includes a new implementation for animated GIFs that is way more performant and shouldn't suffer from memory leaks. Let us know if you're still experiencing problems with 1.4.0.
